src/test/scala/nl/hugo/redbook/ch6/Test6_03.scala

```diff
@@ -1,81 +1,92 @@
+  "intDouble" should {
 
-      RNG.intDouble(mock)._1 should be((1234, toDouble(5678)))
+    "return random int-double tuples" in {
+      var rng: RNG = RNG.Simple(System.currentTimeMillis)
+      for (_ <- 1 to 1000) {
+        val ((n, d), r) = RNG.intDouble(rng)
+        rng = r
+        n should be >= 0
+        d should be >= 0.0
+        d should be < 1.0
+      }
     }
 
-    "convert zeros to zeros" in {
-      val mock = RNGMock(0, RNGMock(0, rng))
-
-      RNG.intDouble(mock)._1 should be((0, 0.0))
+    "return a new generator" in {
+      val rng = CNG(0)
+      val (_, r) = RNG.intDouble(rng)
+      r should be(last)
     }
 
-    "return a negative integer value" in {
-      val mock = RNGMock(-100, RNGMock(0, rng))
-
-      RNG.intDouble(mock)._1 should be((-100, 0.0))
-    }
-
-    "return the next RNG object" in {
-      val mock = RNGMock(0, RNGMock(1, rng))
-
-      RNG.intDouble(mock)._2 should be(rng)
+    "not use the same generator twice" in {
+      val rng = CNG(7, 6)(CNG.zero)
+      val (_, r) = RNG.intDouble(rng)
+      r should be(CNG.zero)
     }
   }
 
-  "RNG.doubleInt" should {
-    "convert two integers into a double and an integer" in {
-      val mock = RNGMock(1234, RNGMock(5678, rng))
-
-      RNG.doubleInt(mock)._1 should be((toDouble(1234), 5678))
-    }
-
-    "convert zeros to zeros" in {
-      val mock = RNGMock(0, RNGMock(0, rng))
-
-      RNG.doubleInt(mock)._1 should be((0.0, 0))
+  "doubleInt" should {
+
+    "return random double-int tuples" in {
+      var rng: RNG = RNG.Simple(System.currentTimeMillis)
+      for (_ <- 1 to 1000) {
+        val ((d, n), r) = RNG.doubleInt(rng)
+        rng = r
+        d should be >= 0.0
+        d should be < 1.0
+        n should be >= 0
+      }
     }
 
-    "return a negative integer value" in {
-      val mock = RNGMock(0, RNGMock(-100, rng))
-
-      RNG.doubleInt(mock)._1 should be((0.0, -100))
+    "return a new generator" in {
+      val rng = CNG(0)
+      val (_, r) = RNG.doubleInt(rng)
+      r should be(last)
     }
 
-    "return the next RNG object" in {
-      val mock = RNGMock(0, RNGMock(1, rng))
-
-      RNG.doubleInt(mock)._2 should be(rng)
+    "not use the same generator twice" in {
+      val rng = CNG(7, 6)
+      val (_, r) = RNG.doubleInt(rng)
+      r should be(last)
     }
   }
 
-  "RNG.double3" should {
-    "convert three integers into three doubles" in {
-      val mock = RNGMock(123, RNGMock(456, RNGMock(789, rng)))
-
-      RNG.double3(mock)._1 should be((toDouble(123), toDouble(456), toDouble(789)))
+  "double3" should {
+
+    "return random tuples with 3 doubles" in {
+      var rng: RNG = RNG.Simple(0)
+      for (_ <- 1 to 1000) {
+        val ((d1, d2, d3), r) = RNG.double3(rng)
+        rng = r
+        d1 should be >= 0.0
+        d1 should be < 1.0
+        d2 should be >= 0.0
+        d2 should be < 1.0
+        d3 should be >= 0.0
+        d3 should be < 1.0
+        d1 should not be (d2)
```

Comment:
You're right. The checks only work because of the fixed seed. However, checking the expected values is impossible since this would again dictate a certain implementation.